crates/uv-requirements/src/lookahead.rs

```diff
@@ -159,16 +160,28 @@ impl<'a, Context: BuildContext> LookaheadResolver<'a, Context> {
         // buildable distribution.
         let dist = match requirement.source {
             RequirementSource::Registry { .. } => return Ok(None),
-            RequirementSource::Url { url, .. } => Dist::from_http_url(requirement.name, url)?,
-            RequirementSource::Git { url, .. } => Dist::Source(SourceDist::Git(GitSourceDist {
+            RequirementSource::Url {
+                subdirectory,
+                location,
+                url,
+            } => Dist::from_http_url(requirement.name, location, subdirectory, url)?,
+            RequirementSource::Git {
+                repository,
+                reference,
+                subdirectory,
+                url,
+            } => Dist::Source(SourceDist::Git(GitSourceDist {
                 name: requirement.name,
+                git: Box::new(GitUrl::new(repository, reference)),
+                subdirectory,
                 url,
             })),
             RequirementSource::Path {
-                path: _,
+                path,
                 url,
+                // TODO(konsti): Figure out why we lose the editable here (does it matter?)
```

Comment:
This was introduced with https://github.com/astral-sh/uv/commit/cf3093283100ef7ff916ce2b7e9cc0061adb2886#diff-49326157b193e4affae1b7a1c56c7c7e1ea2039168e33948b6c30a4fe7f4ef9eR76, where the `Dist::from_url` dropped the information. Is there an argument that the lookahead should or shouldn't consider editables?

Reply:
I think it's because we currently assume editables can't be transitive deps.